### PR TITLE
303 system lapse dialog

### PR DESCRIPTION
**Pull request by wbaccinelli**
Fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/303

The needed components for getting and analyzing the sensors data have been moved to the `sensor_api`, as this has to be used both in the `scheduler` and in the `rasa_actions`

### CODE REVIEW

File: scheduler/celery_utils.py [hunk 1]
```diff
@@ -64,6 +66,42 @@ def check_if_user_active(user_id: int, current_date: date, days_number) -> bool:
...
+        if len(steps_taken) < 2:
+            relapse = True
+        else:
+            # get the days where the goal has not been reached
+            not_reached = [entry for entry in steps_taken if
+                           entry['steps'] < get_current_steps_goal(user_id, entry['date'])]
+            # if goal not reached 4 days or 3 days in a row
+            if len(not_reached) >= 4 or (
+                    len(not_reached) == 3 and not_reached[-1] == range_start - timedelta(days=2)):
+                relapse = True
```
[the remark this answers is not in the document]
Comment (bscheltinga) [reply scheduler/celery_utils.py:100]: Also, this code queries the db 6 times, with a lot of overlapping data, which is probably inefficient. Maybe it is more efficient to do it all with one query, as done in: https://github.com/PerfectFit-project/virtual-coach-rasa/blob/f47c5cc3e94d850d7a537883e2f5c9e39b5c3ca1/Rasa_Bot/actions/helper.py#L1426

File: scheduler/requirements.txt [hunk 2]
```diff
@@ -1,7 +1,9 @@
 celery
 Django==4.2.2
+numpy
 python-dotenv
-git+https://github.com/PerfectFit-project/virtual-coach-db@v0.1.21
+git+https://github.com/PerfectFit-project/virtual-coach-db@303-system-lapse-dialog
 git+https://github.com/PerfectFit-project/niceday_client@v0.1.4
+git+https://github.com/PerfectFit-project/virtual-coach-rasa@303-system-lapse-dialog#subdirectory=sensor_api
```
Comment (bscheltinga) [inline scheduler/requirements.txt:7]: Mind these ;)

File: scheduler/state_machine/controller.py [hunk 3]
```diff
@@ -522,7 +522,8 @@ def on_dialog_completed(self, dialog):
                       Components.RELAPSE_DIALOG_HRS,
                       Components.RELAPSE_DIALOG_LAPSE,
                       Components.RELAPSE_DIALOG_RELAPSE,
-                      Components.RELAPSE_DIALOG_PA]:
+                      Components.RELAPSE_DIALOG_PA,
+                      Components.RELAPSE_DIALOG_SYSTEM]:
 
             logging.info('Relapse dialog completed ')
```
Comment (bscheltinga) [inline scheduler/state_machine/controller.py:528]: A debug remainder?

File: scheduler/celery_tasks.py [hunk 4]
```diff
@@ -57,12 +59,41 @@ def setup_periodic_tasks(sender, **kwargs):  # pylint: disable=unused-argument
     sender.add_periodic_task(crontab(hour=00, minute=00), notify_new_day.s())
     # check if the user is active and send notification
     sender.add_periodic_task(crontab(hour=10, minute=00), check_inactivity.s())
+    # check if the user is in physical relapse
+    sender.add_periodic_task(crontab(hour=10, minute=00), check_physical_relapse.s())
```
[the remark this answers is not in the document]
Comment (wbaccinelli) [reply scheduler/celery_tasks.py:63]: This runs every 24h at 10

File: Rasa_Bot/config.yml [hunk 5]
```diff
@@ -26,7 +26,7 @@ pipeline:
   - name: ResponseSelector
     epochs: 100
   - name: FallbackClassifier
-    threshold: 0.3  # NB: threshold for low NLU confidence. Need to tune carefully.
+    threshold: 0.7  # NB: threshold for low NLU confidence. Need to tune carefully.
```
[the remark this answers is not in the document]
Comment (wbaccinelli) [reply Rasa_Bot/config.yml:29]: this is permanent. The nlu file has been polished from all the unused intents and the intent recognition improve a lot. The result was that with a low threshold even random words were falling into an intent.

File: scheduler/celery_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wbaccinelli) [reply scheduler/celery_utils.py:100]: To be adapted using the `get_step_goals_and_steps` function after merging

**Pull request merged**